### PR TITLE
Add appname to logged json object

### PR DESCRIPTION
**Pull request by chrisblatchley**
As far I could tell there was no way to add in an additional field to the logs, other than tagging which isn't right for this problem. I want to be able to log from multiple apps to the same ELK server, and which app is sending the logs is obviously important. I may be missing a more architecturally sound way to do this, however.

### CODE REVIEW

**Comment by dwbutler** [comment]
I'd like to keep this gem logging only the standard logstash event fields, if possible. Also, it's designed to be used either with Rails or without.

Maybe what's needed is a way to globally add custom fields every time an event is logged? Maybe by hooking into the formatter somehow?

Speculative interface:

```ruby
LogStashLogger.config do |config|
  config.custom_fields.add("custom_field", -> { business logic })
end
```

**Comment by chrisblatchley** [comment]
Ok, this works for me. Also tests and documentation.

**Comment by akshah123** [comment]
:+1:

**Comment by dwbutler** [comment]
Cool! I'll probably merge and release this soon.

**Comment by dwbutler** [comment]
There is some overhead calling a lambda every time you log a message. I think it would make sense to allow the custom field's value to either be a string or something callable (i.e. responds to `#call`).

**Comment by chrisblatchley** [comment]
Custom fields can be strings now!

File: lib/logstash-logger/formatter.rb [hunk 1]
```diff
@@ -36,6 +36,14 @@ def build_event(message, severity, time)
 
       event['host'] ||= HOST
 
+      LogStashLogger.config.custom_fields.each do |field, msg|
+        if msg.respond_to? (:call)
+          event[field] = msg.call()
+        elsif msg.is_a? String
```
Comment (dwbutler) [inline lib/logstash-logger/formatter.rb]: The explicit check for `String` type probably isn't necessary. I can imagine wanting to pass in a numeric value, or a custom object. As long as the custom value can be converted to JSON, `LogStash::Event` should be happy.

**Comment by dwbutler** [comment]
This is looking pretty cool!

I had another idea and wanted to know what you think of it. Instead of specifying custom fields as a hash, maybe we could allow the configuration to pass the `LogStash::Event` into the block so you can do whatever you'd like with it.

So in configuration you could do something like this:

```ruby
LogStashLogger.configure do |config|
  config.customize_event do |logstash_event|
    logstash_event["app_name"] = "Foo"
  end
end
```

Then, in `LogStashLogger::Formatter`, you would `yield` the event to that block.

This would allow much more flexibility to do whatever you need to do with the `LogStash::Event` object directly.

**Comment by chrisblatchley** [comment]
The one issue I find is that doing something like
```ruby
config = LogStashLogger.config do |conf|
  conf.customize_event do |event|
    event["test1"] = "response1"
    event
  end
end
```
requires you to make sure the last line of the block returns the event object. You could skip that and just merge the hash into the `LogStash::Event.data` hash, but that isn't really better than manually inserting a hash of fields. This makes it a little bit clunky to use. Opinions?

**Comment by dwbutler** [comment]
The last line doesn't need to return the event object. `LogStashLogger::Formatter#build_event` just needs to call the `customize_event` block, passing the `event` as the argument so the block can make modifications. (I mis-spoke in a previous comment about using `yield`. I meant to say the event would be passed to the block.)

**Comment by chrisblatchley** [comment]
Gotcha, give this a try.

**Comment by dwbutler** [comment]
Awesome! I'll pull this in and probably do some minor tweaks, but otherwise it's perfect!

**Comment by dwbutler** [comment]
Thanks again for the contribution!

**Pull request merged**